src/components/Dropdown/DropdownLabel.tsx

```diff
@@ -0,0 +1,124 @@
+import * as React from 'react'
+import * as PropTypes from 'prop-types'
+import * as _ from 'lodash'
+
+import keyboardKey from 'keyboard-key'
+import { ComponentEventHandler, ShorthandValue, Extendable } from '../../../types/utils'
+import { UIComponentProps } from '../../lib/commonPropInterfaces'
+import {
+  customPropTypes,
+  createShorthandFactory,
+  UIComponent,
+  RenderResultConfig,
+  commonPropTypes,
+} from '../../lib'
+import { Image, Icon, Label } from '../..'
+import { IconProps } from '../Icon/Icon'
+
+export interface DropdownLabelProps extends UIComponentProps<DropdownLabelProps, any> {
+  /** The text content for the selected item. */
+  header?: string
+
+  /** The item can have an icon when displayed as selected in a multiple selection. */
+  icon?: ShorthandValue
+
+  /** The item can have an image avatar. */
+  image?: ShorthandValue
+
+  /**
+   * Called when clicking on the label for the selected item, in multiple selection.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props and proposed value.
+   */
+  onClick?: ComponentEventHandler<DropdownLabelProps>
+
+  /**
+   * Called when removing an item from multiple selection.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props and proposed value.
+   */
+  onRemove?: ComponentEventHandler<DropdownLabelProps>
+}
+
+/**
+ * A DropdownLabel is a sub-component of a multiple selection Dropdown. It is used to display the selected
+ * options, as labels, before the input text (if search) or the button (if not search).
+ */
+class DropdownLabel extends UIComponent<Extendable<DropdownLabelProps>, any> {
+  displayName = 'DropdownLabel'
+
+  static create: Function
+
+  className = 'ui-dropdown__label'
+
+  static propTypes = {
+    ...commonPropTypes.createCommon({
+      children: false,
+      content: false,
+    }),
+    header: PropTypes.string,
+    icon: customPropTypes.itemShorthand,
+    image: customPropTypes.itemShorthand,
+    onClick: PropTypes.func,
+    onRemove: PropTypes.func,
+  }
+
+  private handleClick = (e: React.SyntheticEvent) => {
+    _.invoke(this.props, 'onClick', e, this.props)
+  }
+
+  public renderComponent({ classes, styles, rest }: RenderResultConfig<DropdownLabelProps>) {
+    const { header, icon, image } = this.props
+    const optionalImage = !image
```

Comment:
no need for this, the shorthand factory handles undefined arguments; take a look at https://github.com/stardust-ui/react/pull/422/files#r239556574

Reply:
refactored